File: app/src/main/java/org/kinecosystem/kinit/dagger/SchedulerModule.java
package org.kinecosystem.kinit.dagger;

import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import org.kinecosystem.kinit.util.AndroidScheduler;
import org.kinecosystem.kinit.util.Scheduler;

@Module
public class SchedulerModule {

    @Provides
    @Singleton
    public Scheduler scheduler() {
        return new AndroidScheduler();
    }
}

File: app/src/main/java/org/kinecosystem/kinit/dagger/SchedulerModule.kt
package org.kinecosystem.kinit.dagger

import dagger.Module
import dagger.Provides
import org.kinecosystem.kinit.util.AndroidScheduler
import org.kinecosystem.kinit.util.Scheduler
import javax.inject.Singleton


@DebugOpenClass
@Module
class SchedulerModule {

    @Provides
    @Singleton
    fun scheduler(): Scheduler = AndroidScheduler()
}
